File: Backend/services/flight-service/src/models/Flight.ts
import { Schema, model, Document } from 'mongoose';

export interface IFlight extends Document {
  flightNumber: string;
  departureAirport: string;
  arrivalAirport: string;
  departureDate: Date;
  departureTime: string;
  arrivalTime: string;
  availableSeats: number;
  totalSeats: number;
  price: number;
  flightStatus: 'Scheduled' | 'Delayed' | 'Cancelled';
}

const FlightSchema = new Schema<IFlight>({
  flightNumber: { type: String, required: true, uppercase: true, trim: true },
  departureAirport: { type: String, required: true, trim: true },
  arrivalAirport: { type: String, required: true, trim: true },
  departureDate: { type: Date, required: true },
  departureTime: { type: String, required: true, trim: true },
  arrivalTime: { type: String, required: true, trim: true },
  availableSeats: { type: Number, required: true },
  totalSeats: { type: Number, required: true },
  price: { type: Number, required: true },
  flightStatus: { type: String, enum: ['Scheduled', 'Delayed', 'Cancelled'], default: 'Scheduled' }
});

export const Flight = model<IFlight>('Flight', FlightSchema);

File: .Backend-old/SkyLink.FlightService/Models/Flight.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SkyLink.FlightService.Models
{
    public class Flight
    {
        public int Id { get; set; }

        [Required]
        public string FlightNumber { get; set; } = string.Empty;

        [Required]
        public string DepartureAirport { get; set; } = string.Empty;

        [Required]
        public string ArrivalAirport { get; set; } = string.Empty;

        [Required]
        public DateTime DepartureDate { get; set; }

        [Required]
        public string DepartureTime { get; set; } = string.Empty; // e.g. "08:30"

        [Required]
        public string ArrivalTime { get; set; } = string.Empty; // e.g. "11:45"

        [Required]
        public int AvailableSeats { get; set; }

        [Required]
        public int TotalSeats { get; set; }

        [Required]
        [Column(TypeName = "decimal(18,2]")]
        public decimal Price { get; set; }

        [Required]
        public string FlightStatus { get; set; } = "Scheduled"; // Scheduled, Boarding, Delayed, Cancelled, Completed
    }
}
